app/uk/gov/hmrc/eoricommoncomponent/frontend/forms/models/mappings/Constraints.scala

```diff
@@ -16,34 +16,167 @@
 
 package uk.gov.hmrc.eoricommoncomponent.frontend.forms.models.mappings
 
-import play.api.data.validation.{Constraint, Invalid, Valid}
+import play.api.data.validation.{Constraint, Invalid, Valid, ValidationError}
+import uk.gov.hmrc.domain.Nino
+import uk.gov.hmrc.eoricommoncomponent.frontend.forms.FormUtils.formatInput
 
 import java.time.LocalDate
+import scala.util.matching.Regex
 
 trait Constraints {
 
-  protected def firstError[A](constraints: Constraint[A]*): Constraint[A] =
-    Constraint { input =>
-      constraints
-        .map(_.apply(input))
-        .find(_ != Valid)
-        .getOrElse(Valid)
-    }
+  protected val noTagsRegex = "^[^<>]+$"
```

Comment:
please refer to the comment above , 
ETMP has an issue with special characters (`#`) in address, maybe we should use the same Regex mentioned above.